bdb/fetch.c

```diff
@@ -735,6 +735,10 @@ static int bdb_fetch_int_ll(
             }
 
             dbp = bdb_state->dbp_data[0][dtafile];
+            extern int gbl_reorder_idx_writes;
+            if (args->for_write && gbl_reorder_idx_writes) {
```

Comment:
Yup- looks correct .. might even be beneficial to do this even if gbl_reorder_idx_writes is not enabled, but I think we intend to keep gbl_reorder_idx_writes as the default ..